test/hotspot/jtreg/testlibrary_tests/template_framework/examples/TestTutorial.java

```diff
@@ -0,0 +1,539 @@
+import static compiler.lib.template_framework.Template.$;
+import static compiler.lib.template_framework.Template.fuel;
+import static compiler.lib.template_framework.Template.addName;
+import static compiler.lib.template_framework.Template.sampleName;
+import static compiler.lib.template_framework.Template.weighNames;
+
+import compiler.lib.template_framework.library.Hooks;
+
+public class TestTutorial {
+
+    public static void main(String[] args) {
+        // Create a new CompileFramework instance.
+        CompileFramework comp = new CompileFramework();
+
+        // Add java source files.
+        comp.addJavaSourceCode("p.xyz.InnerTest1", generateWithListOfTokens());
+        comp.addJavaSourceCode("p.xyz.InnerTest2", generateWithTemplateArguments());
+        comp.addJavaSourceCode("p.xyz.InnerTest3", generateWithHashtagAndDollarReplacements());
+        comp.addJavaSourceCode("p.xyz.InnerTest4", generateWithCustomHooks());
+        comp.addJavaSourceCode("p.xyz.InnerTest5", generateWithLibraryHooks());
+        comp.addJavaSourceCode("p.xyz.InnerTest6", generateWithRecursionAndBindingsAndFuel());
+        comp.addJavaSourceCode("p.xyz.InnerTest7", generateWithNames());
+
+        // Compile the source files.
+        comp.compile();
+
+        // Object ret = p.xyz.InnterTest1.main();
+        comp.invoke("p.xyz.InnerTest1", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest2", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest3", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest4", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest5", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest6", "main", new Object[] {});
+        comp.invoke("p.xyz.InnerTest7", "main", new Object[] {});
+    }
+
+    // This example shows the use of various Tokens.
+    public static String generateWithListOfTokens() {
+        // A Template is essencially a function / lambda that produces a
+        // token body, which is a list of Tokens that are concatenated.
+        var templateClass = Template.make(() -> body(
+            // The "body" method is filled by a sequence of "Tokens".
+            // This can be Strings and multi-line Strings, but also
+            // boxed primitives.
+            """
+            package p.xyz;
+
+            public class InnerTest1 {
+                public static void main() {
+                    System.out.println("Hello World!");
+            """,
+            "int a = ", Integer.valueOf(1), ";\n",
+            "float b = ", Float.valueOf(1.5f), ";\n",
+            // Special Float values are "smartly" formatted!
+            "float nan = ", Float.valueOf(Float.POSITIVE_INFINITY), ";\n",
+            "boolean c = ", Boolean.valueOf(true), ";\n",
+            // Lists of Tokens are also allowed:
+            List.of("int ", "d = 5", ";\n"),
+            // That can be great for streaming / mapping over an existing list:
```

Comment:
By "that" you just mean the following line? Maybe rephrase to: "We can also stream / map over an existing list or one created on the fly:

Reply:
haha, now we kinda removed the list, since we are doing stream direclty. I think I will revert your suggestion here, back to `List.of().stream`, just to make clear that we can do all of that.

Reply:
haha, now we kinda removed the list, since we are doing stream direclty. I think I will revert your suggestion here, back to `List.of().stream`, just to make clear that we can do all of that.

Reply:
I like your suggestion, applied!